### ISSUE TITLE
No way to reset Changes (Green added files)

### ISSUE THREAD

**Issue by markp2** [desktop/desktop#7717]
## Description
GitHub Desktop shows files as New and changed (green square), and there is no way to remove them. git log command line shows no added files, just untracked.

![image](https://user-images.githubusercontent.com/33937982/59061744-16c81d80-889c-11e9-81e1-4e2baa8fa1bf.png)

![image](https://user-images.githubusercontent.com/33937982/59065204-159aee80-88a4-11e9-9d8a-dbd1fbc424fc.png)

## Version
* GitHub Desktop: 2.03
* Operating system: macOS 10.14.5 (18F132)

## Steps to Reproduce
I believe the issue was caused when I used Github to create an automatic stash, and then unstash  it. From that moment on GitHub Desktop refused to stop tracking a bunch of files.

### Expected Behavior
A way to remove these like git reset.

### Actual Behavior
I have spend hours trying to remove the files. If I move them elsewhere, then they disappear. If If I discard changes the files are then DELETED from disk.
If I restore the files, even to older version, they still show up as changed green files.
Basically GitHub Desktop has decided to track these files and whatever I do I cannot remove them.

## Additional Information
Moving the files and folders into a different location removes them for the Changes list.
Renaming the folder has no effect, GitHub Desktop continues to track the new names.

**Comment by tierninho** [comment]
@markp2 Thanks for filing this issue and hope you can get unblocked soon.

Try:
`git clean -df` then `git checkout --`

Note: `git clean` removes all untracked files and `git checkout` clears all unstaged changes. You’ll want to be pretty careful with `git clean`, since it’s designed to remove files from your working directory that are not tracked. If you change your mind, there is often no retrieving the content of those files. 

Here are my logs so you can see what I did:

```
➜  dsgsdg git:(bbb) git status
On branch bbb
Your branch is up to date with 'origin/bbb'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	New Rich Text Document copy.rtf
	ggg copy.md
	sdfsdf copy
	~$w Rich Text Document copy.rtf

nothing added to commit but untracked files present (use "git add" to track)
➜  dsgsdg git:(bbb) ✗ git clean -df  
Removing New Rich Text Document copy.rtf
Removing ggg copy.md
Removing sdfsdf copy
Removing ~$w Rich Text Document copy.rtf
➜  dsgsdg git:(bbb) git checkout --
Your branch is up to date with 'origin/bbb'.
➜  dsgsdg git:(bbb) 
```

**Comment by markp2** [comment]
@tierninho Thanks.

However, I need those untracked files!

I guess I don't understand why GitHub Desktop is insisting on tracking the files which I wish to be **untracked**. git command line is very happy with the untracked files?

- Is there an equivalent of `git reset` for Github desktop?
- Is there a way to clear GitHub desktops cache?

**Comment by iAmWillShepherd** [comment]
@markp2 

> I guess I don't understand why GitHub Desktop is insisting on tracking the files which I wish to be untracked. git command line is very happy with the untracked files?

Desktop is not tracking those files. The green squares you are seeing indicates that Desktop has new (untracked) files. 

<img width="257" alt="Screen Shot 2019-06-06 at 4 52 15 PM" src="https://user-images.githubusercontent.com/1715082/59069189-14ee6200-887c-11e9-8aad-39850a956ded.png">

☝️ Amber squares indicate that a file is being tracked.

If you'd like to Git to not show these files in status, you can 
* Delete them*
* Reset your branch*, `git reset --hard HEAD` (Desktop just presents the parsed output of `git status`) 
* Stash them for later

_`*` indicates you will lose your work_

**Comment by iAmWillShepherd** [comment]
@markp2 

> Is there a way to clear GitHub desktops cache?

Desktop does not have a stash. It parses the output of `git status` to present the changes you see in the UI.

**Comment by iAmWillShepherd** [comment]
@markp2

> git command line is very happy with the untracked files?

Desktop is just as happy with these untracked files as well. You can keep them around for as long as you like. If those files are blocking you from making commits, just unstage them first by de-selecting the checkmark next to the files you wish to keep untracked and you're good to go 🥇.

**Comment by markp2** [comment]
@iAmWillShepherd

Thanks, so is this something new in Desktop? I have had untracked files forever and they have never appeared in Changes? I have always had the files show up in `git status` but they have not caused an issue in desktop. The problem all started when I  used desktop stash instead of command line  stash for the first time – and it ended up stashing/restoring all these untracked files which got me into this state.

Furthermore, I now get a very annoying dialog every time I attempt to switch branch because of these changes (new files) desktop thinks I have?

![image](https://user-images.githubusercontent.com/33937982/59069856-75979200-88b0-11e9-83cd-6c7e7dfdb879.png)

**Comment by markp2** [comment]
@iAmWillShepherd 

>  just unstage them first by de-selecting the checkmark next to the files you wish to keep untracked 

Thanks, but even if I do that, desktop insists  I have changes:

![image](https://user-images.githubusercontent.com/33937982/59070161-a1674780-88b1-11e9-836e-92e0331fd621.png)

**Comment by iAmWillShepherd** [comment]
> Thanks, but even if I do that, desktop insists I have changes:

That is correct. You do still have changes. If you run `git status` on your repo, Git will show that you have untracked changes as well, so Desktop is giving you the same info. You must do something with these changes since Git does its best to prevent you from losing work. 

I suspect, by looking at your files, that a majority of those changes are autogenerated as a result of some build or deployment process. If my assumption is correct, adding those files/directories to your `.gitignore` file should prevent Git from tracking future changes to these files.

I'm going to close this issue out since this is not a bug and how Git works.
